power-sequencer/mihawk-cpld.cpp

```diff
@@ -596,5 +682,65 @@ void MIHAWKCPLD::clearCPLDregister()
     close(fd);
 }
 
+// Check for Mowgli's HDDFault
+int MIHAWKCPLD::checkFault(int reg)
```

Comment:
it would be a little cleaner to just have this function return an HDDRebuildCode instead of having to cast it later?

Reply:
OK, I have adjusted this function to make it return result, and determine which event to report based on the returned result.
I have removed the part of having to cast result.

Reply:
I just realized that you use checkFault() to return two different enum types, depending on the register.  Since that is the case, I think it is better how you had it originally so you can use the enums in the switch statement.  Sorry about that.

Reply:
I have modified it to use the enums originally in the switch statement.
But I'm confused with " use checkFault() to return two different enum types"...
Do you mean that using one function(checkFault()) to return two different enum types of data (HDDErrorCode and HDDRebuildCode)?
My understanding is to add two corresponding types of functions to return the HDDErrorCode and HDDRebuildCode data respectively.
If my understanding is wrong please free to let me know.

Reply:
Yea, I just meant that since that function returns either hdd error or hdd rebuild errors, it couldn't return just a single enum type.  I like your solution on how you refactored it.  Looks good.

Reply:
Thanks for your reviewing.